datashuttle/utils_mod/canonical_configs.py

```diff
@@ -0,0 +1,188 @@
+"""
+This module contains all information for the required
+format of the configs class. This is clearly defined
+as configs can be provided from file or input dynamically
+and so careful checks must be done.
+
+If adding a new config, first add the key to
+get_canonical_config_dict( and type to
+get_canonical_config_required_types()
+"""
+
+from pathlib import Path
+from typing import Union, get_args
+
+from datashuttle.utils_mod import canonical_configs, utils
+
+
+def get_canonical_config_dict():
+    """
+    The only permitted keys in the
+    DataShuttle config.
+    """
+    config_dict = {
+        "local_path": None,
+        "remote_path": None,
+        "connection_method": None,
+        "remote_host_id": None,
+        "remote_host_username": None,
+        "use_ephys": None,
+        "use_behav": None,
+        "use_imaging": None,
+        "use_histology": None,
+    }
+    return config_dict
+
+
+def get_experiment_types():
+    return ["use_ephys", "use_behav", "use_imaging", "use_histology"]
+
+
+def get_canonical_config_required_types():
+    """
+    The only permitted types for DataShuttle
+    config values.
+    """
+    required_types = {
+        "local_path": Union[str, Path],
+        "remote_path": Union[str, Path],
+        "connection_method": str,
```

Comment:
because this is not any string but only takes one of two specific string values, I think you can use `typing.Literal`

Reply:
But double-check the above, I'm not sure I've got the usage of `Literal` 100%

Reply:
That's awesome!